tests/Microcks.Testcontainers.Tests/MicrocksContractTestingFunctionalityTests.cs

```diff
@@ -114,6 +114,20 @@ public async Task ShouldReturnSuccess_WhenGoodImplementation()
         Assert.Equal(3, badTestResult.TestCaseResults.Count);
         Assert.Contains("string found, number expected", badTestResult.TestCaseResults[0].TestStepResults[0].Message);
 
+        // Retrieve messages for the failing test case.
+        List<RequestResponsePair> messages = await _microcksContainer.GetMessagesForTestCaseAsync(badTestResult, "GET /pastries");
+        Assert.Equal(3, messages.Count);
+        for (int i = 0; i < messages.Count; i++)
+        {
+            Assert.NotNull(messages[i].Request);
+            Assert.NotNull(messages[i].Response);
+            Assert.NotNull(messages[i].Response.Content);
+            // Check these are the correct requests.
+            Assert.NotNull(messages[i].Request.QueryParameters);
+            Assert.Single(messages[i].Request.QueryParameters);
+            Assert.Equal("size", messages[i].Request.QueryParameters[0].Name);
+        }
```

Comment:
I think you can write that. It's not a big improvement, you can keep your code.
```suggestion
        Assert.All(messages, message =>
        {
          Assert.NotNull(message.Request);
          Assert.NotNull(message.Response);
          Assert.NotNull(message.Response.Content);
          // Check these are the correct requests.
          Assert.NotNull(message.Request.QueryParameters);
          Assert.Single(message.Request.QueryParameters);
          Assert.Equal("size", message.Request.QueryParameters[0].Name);
        });
```

Reply:
Resolved in new commit